src/site/content/en/fast/codelab-adapt-video-to-image-serving-based-on-network-quality/index.md

```diff
@@ -0,0 +1,210 @@
+---
+layout: codelab
+title: Adapt video to image serving based on network quality
+authors:
+  - mihajlija
+description: |
+  Learn how to use the Network Information API to adapt
+  your content based on the network quality.
+glitch: adaptive-serving-netinfo-starter
+related_post: adaptive-serving-based-on-network-quality
+---
+
+In this codelab, you’ll learn how to adapt your content based on the network
+quality. This page's background video will load only when users are on a fast
+network. On slower networks, an image will load instead.
+
+The
+[Network Information API](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation)
+enables you to access information about the user's connection quality. You will
+use its `effectiveType` property to decide when to serve a video and when to
+serve an image. `effectiveType` can be `'slow-2g'`, `'2g'`, `'3g'`, or `'4g'`.
+
+## Step 1: Check connection type
+
+The background video is currently specified on line 22 of `index.html` and then
```

Comment:
Also: since there's a longer explanation of this later on, maybe have a reference here, like, "The video loading code is described in more detail in [section]"